src/main/java/uk/gov/mca/beacons/service/registrations/RegistrationsService.java

```diff
@@ -1,27 +1,98 @@
 import uk.gov.mca.beacons.service.model.Registration;
+import uk.gov.mca.beacons.service.repository.BeaconPersonRepository;
+import uk.gov.mca.beacons.service.repository.BeaconRepository;
+import uk.gov.mca.beacons.service.repository.BeaconUseRepository;
 
 @Service
+@Transactional
 public class RegistrationsService {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(
     RegistrationsService.class
   );
 
-  public Registration register(Registration registration) {
-    // TODO: Persist beacon, owner etc. to database, send confirmation email
+  private final BeaconRepository beaconRepository;
+  private final BeaconUseRepository beaconUseRepository;
+  private final BeaconPersonRepository beaconPersonRepository;
+
+  @Autowired
+  public RegistrationsService(
+    BeaconRepository beaconRepository,
+    BeaconUseRepository beaconUseRepository,
+    BeaconPersonRepository beaconPersonRepository
+  ) {
+    this.beaconRepository = beaconRepository;
+    this.beaconPersonRepository = beaconPersonRepository;
+    this.beaconUseRepository = beaconUseRepository;
+  }
 
+  public Registration register(Registration registration) {
     LOGGER.info("Attempting to persist registration {}", registration);
 
-    // Create new BeaconPerson to represent owner, persist
-    // For each beacon posted, create new Beacon and persist
-    // -----> For each use associated with Beacon, create new BeaconUse and persist
-    // For each beacon posted, create new Beacon and persist
+    final List<Beacon> beacons = registration.getBeacons();
+    beacons.forEach(this::registerBeacon);
 
     return registration;
   }
+
+  private void registerBeacon(Beacon beacon) {
+    beacon.setBeaconStatus(BeaconStatus.NEW);
+    final Beacon persistedBeacon = beaconRepository.save(beacon);
+
+    registerBeaconUses(persistedBeacon);
+    registerOwner(persistedBeacon);
+    registerEmergencyContacts(persistedBeacon);
+  }
+
+  private void registerBeaconUses(Beacon beacon) {
+    final List<BeaconUse> beaconUses = beacon.getUses();
+    final UUID beaconId = beacon.getId();
+
+    beaconUses.forEach(
+      use -> {
+        use.setBeaconId(beaconId);
+        beaconUseRepository.save(use);
+      }
+    );
+  }
+
+  private void registerOwner(Beacon beacon) {
+    final BeaconPerson owner = beacon.getOwner();
+    final UUID beaconId = beacon.getId();
+
+    registerPerson(owner, beaconId, PersonType.OWNER);
+  }
+
+  private void registerEmergencyContacts(Beacon beacon) {
+    final List<BeaconPerson> emergencyContacts = beacon.getEmergencyContacts();
+    final UUID beaconId = beacon.getId();
+
+    emergencyContacts.forEach(
+      person -> registerPerson(person, beaconId, PersonType.EMERGENCY_CONTACT)
+    );
+  }
+
+  private void registerPerson(
+    BeaconPerson person,
+    UUID beaconId,
+    PersonType personType
+  ) {
+    person.setBeaconId(beaconId);
+    person.setPersonType(personType);
+    beaconPersonRepository.save(person);
+  }
```

Comment:
Does `person.setBeaconId()` mean a `one-to-one` relationship between a `BeaconPerson` and a `Beacon`?